### PR TITLE
[Plugin] ChangeRingtone

### PR DESCRIPTION
**Pull request by FuroYT**
Allows you to change the incoming (default and rare) and outgoing call ringtones using direct URLs to an mp3 file
Note: Feel free to correct any typos that you see (or even let me know) as i am not an Native English speaker

### CODE REVIEW

File: src/plugins/changeRingtone/index.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { definePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+
+const settings = definePluginSettings({
+    incommingRingtoneURL: {
+        description: "Incomming Call Ringtone URL",
```
Comment (schoeneBiene) [inline src/plugins/changeRingtone/index.ts:13]:
```suggestion
    incomingRingtoneURL: {
        description: "Incoming Call Ringtone URL",
```

File: src/plugins/changeRingtone/index.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            replacement: {
+                match: "e.exports=n.p+\"99b1d8a6fe0b95e99827.mp3\"",
+                replace: "e.exports=Vencord.Settings.plugins.ChangeRingtone.incommingRingtoneURL"
+            },
+        },
+        {
+            find: "986703daecf955ce3ce3",
+            replacement: {
+                match: "e.exports=n.p+\"986703daecf955ce3ce3.mp3\"",
+                replace: "e.exports=Vencord.Settings.plugins.ChangeRingtone.incommingRingtoneURL"
```
Comment (schoeneBiene) [inline src/plugins/changeRingtone/index.ts:45]:
```suggestion
                replace: "e.exports=Vencord.Settings.plugins.ChangeRingtone.incomingRingtoneURL"
            },
        },
        {
            find: "986703daecf955ce3ce3",
            replacement: {
                match: "e.exports=n.p+\"986703daecf955ce3ce3.mp3\"",
                replace: "e.exports=Vencord.Settings.plugins.ChangeRingtone.incomingRingtoneURL"
```

File: src/plugins/changeRingtone/index.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+
+const settings = definePluginSettings({
+    incommingRingtoneURL: {
+        description: "Incomming Call Ringtone URL",
+        default: "https://discord.com/assets/986703daecf955ce3ce3.mp3",
+        placeholder: "https://discord.com/assets/986703daecf955ce3ce3.mp3",
+        type: OptionType.STRING,
+        restartNeeded: true
```
Comment (schoeneBiene) [inline src/plugins/changeRingtone/index.ts:17]: Are you sure this needs a restart?

**Review by schoeneBiene — COMMENTED** [review]
You should add a README to your plugin

**Comment by Vendicated** [comment]
there's a more generic sound changer pr which this is a duplicate of

**Pull request closed without merging**